src/communication/network_interface_sr.c

```diff
@@ -10849,6 +10849,32 @@ sflashback_get_summary (THREAD_ENTRY * thread_p, unsigned int rid, char *request
   return;
 }
 
+static char *
+packing_loginfos (THREAD_ENTRY * thread_p, char *ptr, FLASHBACK_LOGINFO_CONTEXT context)
+{
+  CDC_LOGINFO_ENTRY *entry;
+
+  for (int i = 0; i < context.num_item; i++)
+    {
+      ptr = PTR_ALIGN (ptr, MAX_ALIGNMENT);
+      // *INDENT-OFF*
+      entry = context.loginfo_queue.front ();
```

Comment:
1. deque will be nice to use for log read mechanism. 
but we don't know exact range of LSA to read log records.

I mean, 
We create log items by searching from `start_lsa` to `end_lsa`, but it is difficult to do in the way as above because we only create as many as the number input by the user. 

if user wants 5 log items per one connection, it traverse start_lsa to next start_lsa to make 5 log items.  
start_lsa ----------|next start lsa -------------------end_lsa

if only 5 log items exists within the range, then deque will be nice to use. 

2. pop() function does not returns the element, just remove the element. So, I used front() first then pop().

Reply:
1. I don't understand the reason you explain for me. I think it is not related to the log info count per request. Please leave more detail for me or we can discuss this online first.
2. OK.

Reply:
1. I think it is better to discussing it online. 

- if log items are generated backward, and 10 log items are in one transaction. 
- user wants 5 log items per connection
- Loginfos are : | Log info 1 ~ Log info 5 | Log info 6 ~ Log info 10 |
  - Loginfo # is generated in reverse time order
  - if direction is backward, then it return at first | Loginfo 1 ~ Loginfo 5 | then  | Loginfo6 ~ Loginfo 10 |, using deque.pop_front
  - if direction is forward, then it returns at first | Loginfo 5 ~ Loginfo 1 | then | Loginfo10 ~ Loginfo6 |, using deque.pop_back
     - this case is problem, because user gets Loginfo 5, 4, 3, 2, 1 and then Loginfo 10,9,8,7,6

Reply:
OK. Let me know when you are ready.

Reply:
I understand your feedback. Thanks.